crates/ruff_linter/src/rules/flake8_pyi/rules/custom_type_var_return_type.rs

```diff
@@ -345,39 +375,33 @@ fn remove_first_parameter_annotation(parameters: &Parameters) -> Edit {
     Edit::deletion(first.name().end(), first.end())
 }
 
-fn replace_return_annotation_with_self(self_symbol_binding: String, returns: &Expr) -> Edit {
-    Edit::range_replacement(self_symbol_binding, returns.range())
+fn replace_return_annotation_with_self(self_symbol_binding: &str, returns: &Expr) -> Edit {
+    Edit::range_replacement(self_symbol_binding.to_string(), returns.range())
 }
 
 fn replace_typevar_usages_with_self(
```

Comment:
Can you add some docs to this function? It was quite unclear to me at first why it returns an `Option<Vec<Edit>>` rather than just returning an empty `Vec` in the cases where it currently returns `None`. I think it's because if we can't resolve the binding to something in this module, we can't be sure that the binding refers to a private TypeVar. Is that correct?

It would also be good to explain what `editable_range` represents, and why we only replace references to the TypeVar if the references lie within that range